src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -775,6 +776,7 @@ internal bool TryGetCollectionIterationType(ExpressionSyntax syntax, TypeSymbol
             TypeSymbol targetType,
             BindingDiagnosticBag diagnostics)
         {
+            RuntimeHelpers.EnsureSufficientExecutionStack();
```

Comment:
>RuntimeHelpers.EnsureSufficientExecutionStack(); [](http://example.com/codeflow?start=12&length=48)

It is hard to confirm that all call sites are actually catching the exception right now. It will be even harder to maintain this invariant in the future without some strong hints. I suggest to add a "OrThrowStackOverflow" suffix to the name of this method and any other method calling ```RuntimeHelpers.EnsureSufficientExecutionStack``` explicitly, or calling a method with the suffix and not having try/catch(```InsufficientExecutionStackException```) around the call.